dataactbroker/handlers/fileHandler.py

```diff
@@ -639,28 +639,52 @@ def upload_detached_file(self, create_credentials):
             }
 
             job_data = {}
+            # Check for existing submission
+            existing_submission_id = request_params.get('existing_submission_id')
+            if existing_submission_id:
+                existing_submission = True
+                existing_submission_obj = sess.query(Submission).\
+                    filter_by(submission_id=existing_submission_id).\
+                    one()
+                jobs = sess.query(Job).filter(Job.submission_id == existing_submission_id)
+                # set all jobs to their initial status of "waiting"
+                jobs[0].job_status_id = JOB_STATUS_DICT['waiting']
+                sess.commit()
+
+            else:
+                existing_submission = None
+                existing_submission_obj = None
+
             for request_field, submission_field in request_job_mapping.items():
                 if request_field in request_params:
                     job_data[submission_field] = request_params[request_field]
                 # all of those fields are required
-                else:
+                # unless existing_submission_id exists
+                elif 'existing_submission_id' not in request_params and request_field in request_params:
```

Comment:
If it's always required, couldn't you just add `existing_submission_id` to `request_job_mapping`? Rather than having this strange combination of both?